enterprise-suite/templates/alertmanager.yaml

```diff
@@ -1,3 +1,18 @@
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: alertmanager-storage
+spec:
+  storageClassName: {{ default .Values.defaultStorageClass .Values.esAlertmanagerStorageClass }}
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: {{ .Values.alertmanagerVolumeSize }}
+  parameters:
+    fstype: xfs
```

Comment:
This seems an odd detail to have at this level. What's the idea behind specifying it here instead of upstream in the storage class?

Reply:
As far as I know you can't specify fstype at the StorageClass. fstype is used by kubelet when formatting the volume, rather than the provisioning of storage.

Reply:
Okay take that back - seems like you can put it in either place. But we don't have control of the StorageClass - it's expected users will provide that. So I think it's safer to specify in the PVC, rather than relying on users.

Reply:
But... if this fstype parameter is incompatible with the user defined storage class, what happens? Why would we force it here, instead of recommend it in the docs for certain kinds of StorageClasses? What if a customer wants some cloud provider blob store thingy we never heard of that doesn't support xfs? Does everything fail at this point? Or is our fstype parameter ignored?

On face this looks like a detail that pierces the abstraction: provide us a StorageClass, then we'll mutate it for you so it has xfs...  see what I mean?

Reply:
Maybe, I think you can view it either way. I think the support for fstype on StorageClass is recent. You used to have to specify it on the PersistentVolume. The StorageClass defines the backing storage. File system is independent of that, and the user can define what they want.